### PR TITLE
Fix correct accumulated time for parallelized jobs in SGEN.

### PR DESCRIPTION
**Pull request by lateralusX**
Current implementation incorrectly accumulate time for scan jobs for the follow metrics when running parallel minor GC:

time_minor_scan_major_blocks,
time_minor_scan_los,
time_major_scan_mod_union_blocks,
time_major_scan_mod_union_los

Commit fix so updates are atomic making sure values are correct in cases where parallel minor GC is used.

### CODE REVIEW

File: mono/sgen/sgen-gc.c [hunk 1]
```diff
@@ -1446,7 +1446,9 @@ job_scan_major_card_table (void *worker_data_untyped, SgenThreadPoolJob *job)
 	SGEN_TV_GETTIME (atv);
 	sgen_major_collector.scan_card_table (CARDTABLE_SCAN_GLOBAL, ctx, job_data->job_index, job_data->job_split_count, job_data->data);
 	SGEN_TV_GETTIME (btv);
-	time_minor_scan_major_blocks += SGEN_TV_ELAPSED (atv, btv);
+
+	gint64 elapsed_time = SGEN_TV_ELAPSED (atv, btv);
+	mono_atomic_add_i64 ((volatile gint64 *)&time_minor_scan_major_blocks, elapsed_time);
 
 	if (worker_data_untyped)
 		((WorkerData*)worker_data_untyped)->major_scan_time += SGEN_TV_ELAPSED (atv, btv);
```
Comment (BrzVlad) [inline mono/sgen/sgen-gc.c]: elapsed_time I guess

**Comment by lateralusX** [comment]
@monojenkins rebuild failed

**Comment by lateralusX** [comment]
@monojenkins rebuild failed

**Comment by lateralusX** [comment]
@monojenkins rebuild failed

**Comment by lateralusX** [comment]
@monojenkins rebuild failed

**Comment by lateralusX** [comment]
@BrzVlad As previosly discussed, looks like implement sgen version of mono_atomic_add_i64 when we have BROKEN_64BIT_ATOMICS_INTRINSIC fix the ARMv5, ARMv7 problem. The same issue will apply to the other sgen atomic functions used in case where we define WAPI_NO_ATOMIC_ASM. Currently that is only used in WASM (AFAIK) where we won't have multithread support, but could in theory be used on other platforms. Question is if we should do similar fallback solution for the other atomics used by sgen, mono_atomic_cas_i32, mono_atomic_cas_ptr as well to make sure we won't end up in implementations in atomic.c that uses a lock shared by mutators?

**Pull request merged**